### PR TITLE
Separate tests

### PR DESCRIPTION
**Pull request by ralphtheninja**
This cleans up the scripts a bit. ~~I also wanted everything to be named as tests, since they are actually tests.~~

~~Running tests locally by e.g. `npm test` will test everything _but_ the sauce browser tests, which can be run separately by `npm run test:sauce`.~~

~~Travis runs `npm run test:all` which tests everything.~~

Closes https://github.com/airtap/airtap/issues/140

Will squash.

### CODE REVIEW

**Comment by vweevers** [comment]
Seeing the end result, I think we're moving in the wrong direction (including #134). Normally `npm test` is what runs all tests. A `test:all` script is only confusing. And ideally there's only entry point to the tests, with as much coverage as possible.

So we could also do:

```
"scripts": {
  "test": "standard && npm run dependency-check && cross-env DEBUG=airtap* tape test/index.js",
  "dependency-check": "dependency-check package.json --missing --unused -i dependency-check -i phantomjs -i cross-env -i standard --entry test/index.js --entry test/fixtures/*/*.js --entry frameworks/*.js"
}
```

And skip tests automatically, by adding this to `tape-sauce.js`:

```js
if (!process.env.CI && !auth.username && !auth.key) {
  t.skip('no sauce labs credentials provided')
  return t.end()
} else if (!auth.username || !auth.key) {
  t.fail('incomplete sauce labs credentials provided')
  return t.end()
}
```

I'm on the fence, neither solution is 100% clean.

**Comment by ralphtheninja** [comment]
> And skip tests automatically, by adding this to tape-sauce.js:

Sounds good. I got carried away :)

**Comment by ralphtheninja** [comment]
@vweevers Updated.

**Comment by vweevers** [comment]
Weird, seems like travis built an older commit:

```
$ npm run test:all
npm ERR! missing script: test:all
```

Restarting the build.

File: test/integration/tape-sauce.js [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+    return t.end()
+  }
+
   var config = {
     prj_dir: path.resolve(__dirname, '../fixtures/tape'),
     files: [ path.resolve(__dirname, '../fixtures/tape/test.js') ],
     username: auth.username,
-    concurrency: 5,
     key: auth.key,
+    concurrency: 5,
```
Comment (vweevers) [inline test/integration/tape-sauce.js]: oh that feels goood

**Comment by vweevers** [comment]
@ralphtheninja mind pushing a dummy commit? Something is borked in Travis or the git history

**Pull request merged**